### PR TITLE
Issues/broker dev/prevent nonhttps redirect in webview

### LINKED ISSUE #487: Verify that auth flows cannot be redirected to a non https endpoint
Except for end url

### PR DESCRIPTION
**Pull request by jasoncoolmax**
to fix #487 
1. Checking is added to prevent insecure http redirect in webview;
2. Unit tests are added, which include adding a class *ADTestAuthenticationViewController* in test to mock ADAuthenticationViewController.

### CODE REVIEW

File: ADAL/src/ui/ADWebAuthController.m [hunk 1]
```diff
@@ -290,6 +290,17 @@ - (BOOL)webAuthShouldStartLoadRequest:(NSURLRequest *)request
         return NO;
     }
     
+    // redirecting to non-https url is not allowed
+    if ([requestURL hasPrefix: @"http"] && ![requestURL hasPrefix: @"https"])
+    {
```
Comment (jasoncoolmax) [inline ADAL/src/ui/ADWebAuthController.m]: Should we just put 
if (![requestURL hasPrefix: @"https"])
and move the checking to the end of this function?

There are some subtle difference :P

Comment (RPangrle) [inline ADAL/src/ui/ADWebAuthController.m]: just "if ([requestURL hasPrefix: @"http"])" would be sufficient. I could see wanting to specifically block anything that's not either one of the allowed types, redirect URI scheme or HTTPS.

File: ADAL/tests/ADAuthenticationViewControllerTests.m [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    
+    [self waitSemaphoreWithoutBlockingMainQueue:_dsem];
+    
+}
+
+- (void)waitSemaphoreWithoutBlockingMainQueue:(dispatch_semaphore_t)sem
+{
+    while (dispatch_semaphore_wait(sem, DISPATCH_TIME_NOW))
+    {
+        [[NSRunLoop mainRunLoop] runMode:NSDefaultRunLoopMode beforeDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];
```
Comment (RPangrle) [inline ADAL/tests/ADAuthenticationViewControllerTests.m]: You should be able to use [NSDate distantFuture] here,

**Comment by RPangrle** [comment]
:shipit:

**Pull request merged**